src/commands.ts

```diff
@@ -69,6 +69,30 @@ export function registerCommands() {
         }
     }
 
+    async function updateFolderAccess(fileAccess: FileAccess, uri: Uri) {
+        if (await Operations.updateFolderAccess(fileAccess, uri)) {
+            controller.updateStatusBar(fileAccess); 
+        }
+    }
+
+    async function updateByFileType(uri: Uri, fileAccess: FileAccess) {
+        const fileType = Operations.getFileType(uri);
+        if(fileType === FileType.File) {
+            updateFileAccess(fileAccess);
```

Comment:
If the Context Menu is executed on a `File`, its `Uri` is not being used, and you end up changing the File Access to the `ActiveTextEditor` instead of the selected file in `Explorer`. This happens because the `updateFileAccess` function today uses the `vscode.window.activeTextEditor` (because the command were used for the active file). 

You must change the function, adding a `Uri` parameter, just like the `updateFolderAccess` function you created.